drivers/fuel_gauge/sbs_gauge/emul_sbs_gauge.c

```diff
@@ -220,10 +233,49 @@ static int sbs_gauge_emul_transfer_i2c(const struct emul *target, struct i2c_msg
 	return rc;
 }
 
+static int emul_sbs_fuel_gauge_set_battery_charging(const struct emul *target, uint32_t mV, int mA)
+{
+	struct sbs_gauge_emul_data *data = target->data;
+
+	if (mV == 0 || mA == 0)
+		return -EINVAL;
+
+	data->batt_state.mA = mA;
+	data->batt_state.mV = mV;
```

Comment:
One thought as a follow up effort.  
Add a backend hook to set the charging voltage and current.  The emulator could scheduling a recurring work item and slowly adjust the "real" voltage and current as appropriate.

Reply:
Good idea! I'll make a github issue so it's tracked.